test/btree.js

```diff
@@ -0,0 +1,318 @@
+  test.strictSame(bTree.root[1].child[1].key, 160);
+  test.strictSame(bTree.root[2].child[1].key, 190);
+  test.end();
+});
+
+metatests.test('get / number keys', (test) => {
+  const bTree = new BTree(3);
+
+  bTree
+    .set(20,  { city: 'Abu Dhabi'  })
+    .set(50,  { city: 'Ankara'     })
+    .set(100, { city: 'Jerusalem'  })
+    .set(150, { city: 'Seoul'      })
+    .set(200, { city: 'Kiev'       })
+    .set(0,   { city: 'Luxembourg' })
+    .set(160, { city: 'Minsk'      })
+    .set(180, { city: 'Oslo'       })
+    .set(170, { city: 'Stockholm'  });
+
+  test.strictSame(bTree.get(20),  { city: 'Abu Dhabi'  });
+  test.strictSame(bTree.get(50),  { city: 'Ankara'     });
+  test.strictSame(bTree.get(100), { city: 'Jerusalem'  });
+  test.strictSame(bTree.get(150), { city: 'Seoul'      });
+  test.strictSame(bTree.get(0),   { city: 'Luxembourg' });
+  test.strictSame(bTree.get(170), { city: 'Stockholm'  });
+  test.end();
+});
+
+
+metatests.test('get / string keys', (test) => {
+  const bTree = new BTree(3);
+
+  bTree
+    .set('Abuabi',     { country: 'UAE'  })
+    .set('Ankara',     { country: 'Turkey'     })
+    .set('Jerusalem',  { country: 'Israel'  })
+    .set('Seoul',      { country: 'South Korea'      })
+    .set('Kiev',       { country: 'Ukraine'       })
+    .set('Luxembourg', { country: 'Luxembourg' })
+    .set('Minsk',      { country: 'Belarus'      })
+    .set('Oslo',       { country: 'Norway'       })
+    .set('Stockholm',  { country: 'Sweden'  });
+
+  test.strictSame(bTree.get('Abuabi'),    { country: 'UAE'     });
+  test.strictSame(bTree.get('Ankara'),    { country: 'Turkey'  });
+  test.strictSame(bTree.get('Kiev'),      { country: 'Ukraine' });
+  test.strictSame(bTree.get('Minsk'),     { country: 'Belarus' });
+  test.strictSame(bTree.get('Oslo'),      { country: 'Norway'  });
+  test.strictSame(bTree.get('Stockholm'), { country: 'Sweden'  });
+  test.end();
+});
+
+metatests.test('iterator / numer key', (test) => {
+  const bTree = new BTree(2);
+
+  bTree
+    .set(40,  'G')
+    .set(30,  'F')
+    .set(72,  'M')
+    .set(24,  'E')
+    .set(20,  'D')
+    .set(18,  'B')
+    .set(50,  'I')
+    .set(60,  'L')
+    .set(90,  'N')
+    .set(45,  'H')
+    .set(55,  'K')
+    .set(52,  'J')
+    .set(100, 'O')
+    .set(120, 'P')
+    .set(130, 'Q')
+    .set(140, 'R')
+    .set(19,  'C')
+    .set(-20, 'A');
+
+  let res = [...bTree.iterator()].join(''); // All nodes
+  test.strictSame(res, 'ABCDEFGHIJKLMNOPQR');
+
+  res = [...bTree.iterator(undefined, 10000)].join(''); // All nodes
+  test.strictSame(res, 'ABCDEFGHIJKLMNOPQR');
+
+  res = [...bTree.iterator(-10000)].join(''); // All nodes
+  test.strictSame(res, 'ABCDEFGHIJKLMNOPQR');
+
+  res = [...bTree.iterator(100, 0)].join(''); // Empty
+  test.strictSame(res, '');
+
+  res = [...bTree.iterator(-200, -100)].join(''); // Empty
+  test.strictSame(res, '');
+
+  res = [...bTree.iterator(20, 50)].join('');
+  test.strictSame(res, 'EFGH');
```

Comment:
Why does not this include 'D' if it was added as an element with key 20?

Reply:
I thought we agreed to use [a, b) sort of range?

Reply:
I don't remember when we agreed this, but okay, I'll do so